src/compas/datastructures/mesh/core/operations/substitute.py

```diff
@@ -48,7 +48,8 @@ def mesh_substitute_vertex_in_faces(mesh, old_vkey, new_vkey, fkeys=None):
 
     # apply to all faces if there is none chosen
     if fkeys is None:
-        fkeys = mesh.faces()
+        mesh_cp = copy.deepcopy(mesh)  # ifno deepcopy, next loop is unsafe as we modify face dict while looping on it
+        fkeys = mesh_cp.faces()
```

Comment:
why not just
```python
fkeys = list(mesh.faces())
```

Reply:
Just a bad habit from myself. Of course your suggestion is much more efficient.